crates/ide_assists/src/handlers/extract_struct_from_enum_variant.rs

```diff
@@ -204,7 +206,34 @@ fn create_struct_def(
         make::struct_(enum_.visibility(), variant_name, enum_.generic_param_list(), field_list)
             .clone_for_update();
 
-    // copy attributes
+    // FIXME: Consider making this an actual function somewhere (like in `AttrsOwnerEdit`) after some deliberation
+    let attrs_and_docs = |node: &SyntaxNode| {
+        let mut select_next_ws = false;
+        node.children_with_tokens().filter(move |child| {
+            let accept = match child.kind() {
+                ATTR | COMMENT => {
+                    select_next_ws = true;
+                    return true;
+                }
+                WHITESPACE if select_next_ws => true,
+                _ => false,
+            };
+            select_next_ws = false;
+
+            accept
+        })
+    };
```

Comment:
^ this

Modified from `remove_attrs_and_docs` in `AttrsOwnerEdit`, which is the reasoning behind the potential location.
I can open a separate issue + pr (or just a pr) for this later though.